packages/api/pages/api/auth/forgot-password.ts

```diff
@@ -0,0 +1,39 @@
+import * as z from 'zod';
+import createRoute from '../../../shared/Route';
+import mailer from '../../../shared/mailer';
+import { randomBytes } from 'crypto';
+import { client } from '../../../shared/db';
+import { redirects } from '../../../shared/redirects';
+import { PostForgotPasswordRequestBody } from '@translation/api-types';
+
+export default createRoute()
+  .post<PostForgotPasswordRequestBody, void>({
+    schema: z.object({ email: z.string() }),
+    async handler(req, res) {
+      const user = await client.user.findUnique({
+        where: { email: req.body.email },
+      });
+      if (!user) {
+        return res.notFound();
+      }
+      const { token } = await client.resetPasswordToken.upsert({
+        where: { userId: user.id },
+        update: { expires: Date.now() + 60 * 60 * 1000 },
+        create: {
+          userId: user.id,
+          token: randomBytes(12).toString('hex'),
+          expires: Date.now() + 60 * 60 * 1000,
+        },
+      });
```

Comment:
A user can have multiple tokens, so we can just create a new one each time. This ensures that a token can't be reused after it expires. Otherwise a compromised token could be used by requesting a new reset

Reply:
Done.